packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/DateTrigger.java

```diff
@@ -0,0 +1,61 @@
+package expo.modules.notifications.notifications.triggers;
+
+import android.os.Parcel;
+
+import java.util.Date;
+
+import androidx.annotation.Nullable;
+import expo.modules.notifications.notifications.interfaces.SchedulableNotificationTrigger;
+
+/**
+ * A schedulable trigger representing notification to be scheduled only once at a given moment of time.
+ */
+public class DateTrigger implements SchedulableNotificationTrigger {
+  private Date mTriggerDate;
+
+  public DateTrigger(long timestamp) {
+    mTriggerDate = new Date(timestamp);
+  }
+
+  private DateTrigger(Parcel in) {
+    mTriggerDate = new Date(in.readLong());
+  }
+
+  public Date getTriggerDate() {
+    return mTriggerDate;
+  }
+
+  @Nullable
+  @Override
+  public Date nextTriggerDate() {
+    Date now = new Date();
+
+    if (mTriggerDate.before(now)) {
```

Comment:
with this logic, is it possible to create a notification that never fires (if the trigger date has already passed by the time this code runs)? For example, what if you create a notification in JS that is supposed to trigger immediately by setting a date interval of `new Date()` (or, for that matter, a time interval of 0 seconds)?

Reply:
Yes, it is.

Reply:
OK -- so I guess you would use `presentNotificationAsync` instead if you wanted this functionality. That's fine, let's just make sure to document this clearly so developers understand the expected behavior of each method.

Reply:
That's right! On iOS there is no `presentNotification:` native method, presenting notifications is scheduling them with **nil** trigger. I even planned to do so in `expo-notifications`, but then there wouldn't be a good place to put dismissing notifications (NotificationPresenter is a good place right now).